src/sqlfluff/core/parser/segments/base.py

```diff
@@ -1092,10 +1092,32 @@ def apply_fixes(self, dialect, rule_code, fixes):
 
     def _validate_segment_after_fixes(self, rule_code, dialect, fixes_applied, segment):
         """Checks correctness of new segment against match or parse grammar."""
-        found_error = False
         root_parse_context = RootParseContext(dialect=dialect)
         with root_parse_context as parse_context:
-            if getattr(segment, "match_grammar", None):
+            # Check for parse_grammar first. If both are present, it is more
+            # thorough and complete.
```

Comment:
This change looks bigger than it is. Basically, I swapped the two checks, preferring the `parse_grammar` if present, only falling back to `match_grammar` if there's no `parse_grammar`.

Reply:
This looks really good. The only question I have is why are you removing the found_error stuff?

Reply:
Basically, the `elif` removes the need for it.

Old code:
# Check `match_grammar` if exists.
# If we haven't found an error yet (i.e. using `match_grammar`), check `parse_grammar` if exists.

New code:
# Check `parse_grammar` if exists
# Else check `match_grammar` if exists

Reply:
Okay, makes sense - I suppose parse grammar is always more explicit

Reply:
It’s not so much that it’s more explicit, it’s that that’s what the parser uses:

https://docs.sqlfluff.com/en/stable/architecture.html#stage-3-the-parser
> Each segment can specify a parse_grammar, and a match_grammar. If both a match_grammar and parse_grammar are defined in a segment, match_grammar is used to quickly prune the tree for branches which do not match segments being parsed, and the parse_grammar is then used to refine the branch identified as correct. If only a match_grammar is defined, then it serves the purpose of both pruning and refining.